gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/extractor/FileAwareInputStreamExtractorWithCheckSchema.java

```diff
@@ -66,14 +67,19 @@ protected FileAwareInputStream buildStream(FileSystem fsFromFile) throws DataRec
    * @throws IOException
    */
   protected boolean schemaChecking(FileSystem fsFromFile) throws IOException {
+    if( !this.state.getPropAsBoolean(ReplicationConfiguration.COPY_SCHEMA_CHECK_ENABLED, false) ) {
+      return true;
+    }
     DatumReader<GenericRecord> datumReader = new GenericDatumReader<>();
     DataFileReader<GenericRecord> dataFileReader =
         new DataFileReader(new FsInput(this.file.getFileStatus().getPath(), new Configuration()), datumReader);
```

Comment:
There is no existing generic logic to read schema because the schema need to be read by specific reader. And the next PR should enable schema check for ORC schema as well. But it will be implemented by an if else statement which is also not generic.